contrib/ruby/ext/trilogy-ruby/cext.c

```diff
@@ -49,12 +53,29 @@ static size_t trilogy_memsize(const void *ptr) {
     return memsize;
 }
 
+static void trilogy_ctx_mark(void *ptr)
+{
+    struct trilogy_ctx *ctx = ptr;
+
+    if (RTEST(ctx->io))
```

Comment:
Unnecessary?

Reply:
Mark moveable doesn't check if the field is NULL, IIRC without this it can crash.

Reply:
I was looking at https://github.com/ruby/ruby/blob/1fdaa0666086529b3aae2d509a2e71c4247c3a12/gc.c#L7108 and https://github.com/ruby/ruby/blob/1fdaa0666086529b3aae2d509a2e71c4247c3a12/gc.c#L10395 -- I'm not _100%_ sure it's part of their defined contract, but I imagine it must be true by ossification, even if not originally intended?

Intuitively I just prefer "this slot contains a a possible GC ref", rather than an [even trivial] encoding of "we _might_ need to mark this". 🤷🏻‍♂️